### ISSUE TITLE
NMI in a crashloop

### ISSUE THREAD

**Issue by gnadaban** [kubernetes-sigs/cluster-api-provider-azure#1545]
/kind bug

**What steps did you take and what happened:**
Created a CAPZ cluster and noticed that the NMI pods are in a constant crashloop.

**What did you expect to happen:**
NMI should not crashloop.

**Anything else you would like to add:**
Log from a pod:
```
I0720 16:34:18.943358       1 main.go:61] starting nmi process. Version: v1.6.3. Build date: 2020-09-15-21:07.
I0720 16:34:19.046225       1 crd.go:446] CRD lite informers started
I0720 16:34:19.146391       1 main.go:88] running NMI in namespaced mode: true
I0720 16:34:19.146446       1 nmi.go:53] initializing in managed mode
I0720 16:34:19.146454       1 probes.go:41] initialized health probe on port 8085
I0720 16:34:19.146460       1 probes.go:44] started health probe
F0720 16:34:19.146873       1 probes.go:27] http listen and serve error: listen tcp :8085: bind: address already in use
```

**Environment:**

- cluster-api-provider-azure version: 0.4.15
- Kubernetes version: (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T21:10:45Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.7", GitCommit:"132a687512d7fb058d0f5890f07d4121b3f0a2e2", GitTreeState:"clean", BuildDate:"2021-05-12T12:32:49Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`): default

**Comment by devigned** [comment]
Is it possible you have two instances of aad-pod-identity running? There are some scenarios where bind issues come up: https://github.com/Azure/aad-pod-identity/issues?q=is%3Aissue+address+already+in+use. Either the host has something already running on that port, or possibly, two deployments of aad-pod-identity are running.

**Comment by gnadaban** [comment]
Thanks for the tip @devigned , sure enough, I install the AAD pod identity via Helm chart. Is it possible to disable the CAPZ installed one, or configure which version it deploys?

**Comment by devigned** [comment]
I know we discussed not deploying aad-pod-identity at some point, but I don't recall the specifics. Perhaps, @nader-ziada recalls more clearly.

**Comment by gnadaban** [comment]
It would be great if at least it would be an option to not deploy it. I am not familiar with the codebase, where do the infrastructure components get deployed?

**Comment by devigned** [comment]
The infrastructure components via `clusterctl` by it finding the release artifacts (https://github.com/kubernetes-sigs/cluster-api-provider-azure/releases/tag/v0.5.0) and pulling in `infrastructure-components.yaml`. I don't believe there is a way for use to template sections of that yet. 

@CecileRobertMichon is templating `infrastructure-components.yaml` a thing or is that still relatively static?

**Comment by CecileRobertMichon** [comment]
not currently: https://github.com/kubernetes-sigs/cluster-api/issues/3252

this is the same blocker that was discussed back then https://kubernetes.slack.com/archives/CEX9HENG7/p1607093358351000

**Comment by devigned** [comment]
@gnadaban, we definitely can see the desire to control the version / installation of aad-pod-identity.

**Comment by gnadaban** [comment]
For now I opted to delete the Daemonset after cluster-api init since we need the AAD installed for other dependencies beforehand.

For future consideration, it would be great if a switch could disable installing the AAD.

**Comment by CecileRobertMichon** [comment]
/remove-lifecycle rotten

This should be resolved by #2205
